src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/model/validator/NameSizeValidator.java

```diff
@@ -8,15 +8,16 @@
 import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.LONG_MAX_SIZE;
 
 public class NameSizeValidator implements ConstraintValidator<NameSize, Object> {
+
     public boolean isValid(Object object, ConstraintValidatorContext context) {
         if (object instanceof DataDto dto) {
             return isSizeCorrect(dto.getPartnershipName(), dto.getNameEnding());
-        } else if (object instanceof LimitedPartnershipPatchDto dto) {
-            String nameEnding = dto.getNameEnding() != null ? dto.getNameEnding().getDescription() : "";
+        }
+        if (object instanceof LimitedPartnershipPatchDto dto) {
```

Comment:
Not sure about this change. It will evaluate more expressions than before. Suggested compliant solution uses a Switch statement  - could you use that instead, maybe?